lib/kubeclient/common.rb

```diff
@@ -39,6 +39,12 @@ module ClientMixin
       'fieldSelector' => :field_selector
     }.freeze
 
+    DELETE_ARGUMENTS = {
+      'gracePeriodSeconds' => :grace_period_seconds,
+      'orphanDependents'   => :orphan_dependents,
```

Comment:
Is it wise for kubeclient to hardcode specific options of specific kube versions?
Could we just pass all options to server (after camelizing, something like `.gsub(/_([a-z])/) { |m| m[1].upcase }`)?
If a combination is incompatible and server doesn't give an error, that's IMO a server bug...